src/legacy/server/index_patterns/mixin.ts

```diff
@@ -55,12 +40,18 @@ export function indexPatternsMixin(kbnServer, server) {
    *  @method request.getIndexPatternsService
    *  @type {IndexPatternsService}
    */
-  server.addMemoizedFactoryToRequest('getIndexPatternsService', request => {
-    const { callWithRequest } = request.server.plugins.elasticsearch.getCluster('data');
-    const callCluster = (...args) => callWithRequest(request, ...args);
-    return server.indexPatternsServiceFactory({ callCluster });
-  });
+  (server as any).addMemoizedFactoryToRequest(
+    'getIndexPatternsService',
+    (request: Legacy.Request) => {
+      const { callWithRequest } = (request as any).server.plugins.elasticsearch.getCluster('data');
+      const callCluster = (...args: any) => callWithRequest(request, ...args);
+      return server.indexPatternsServiceFactory({ callCluster });
+    }
+  );
 
-  server.route(createFieldsForWildcardRoute(pre));
-  server.route(createFieldsForTimePatternRoute(pre));
+  registerRoutes(kbnServer.newPlatform.setup.core);
 }
+
+export type IndexPatternsServiceFactory = (args: {
+  callCluster: (endpoint: string, clientParams: any, options: any) => Promise<any>;
```

Comment:
```suggestion
  callCluster: APICaller;
```

I know you took this type from an existing file but this actually should be a better type for this.

Reply:
We cannot use `APICaller` here. It breaks type check. In [x-pack/infra](https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/infra/server/lib/adapters/framework/kibana_framework_adapter.ts#L143) and [x-pack/siem](https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/siem/server/lib/framework/kibana_framework_adapter.ts#L112), `callCluster` is called with `GenericParams` which is not compatible with Record<string, unknown>. 

Or should I fix infra and siem codes?